0000-terraform-state-migration.md

```diff
@@ -0,0 +1,107 @@
+S3 backend.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+- As mentioned above, the migrator is a special Docker image that is
+run by a Kubernetes job. It will rely on certain ENV's to be avaiable, with
+information about S3 bucket, Workspace name, Namespace, etc.
+- The special Docker image is a Go app, that utilizes the AWS SDK to pull from
+and push to S3 buckets.
+- The on-boarding customer will supply the following as ENV's defined in the
+Job yaml spec:
+    - S3_BUCKET: the S3 bucket where the on-boarding customer's Terraform state
+    resides
+    - S3_KEY: the key to the above Terraform state
+    - Namespace: the Namespace created in Scipian from the on-boarding customer
+    - Workspace: the name of the Workspace of the on-boarding customer's
+    Terraform
+    - ONBOARDING_AWS_ACCESS_KEY_ID: the access key to the on-boarding customer's
+    AWS account where the state will be pulled from
+    - ONBOARDING_AWS_SECRET_ACCESS_KEY: secret access key to the above account
+    - SCIPIAN_AWS_ACCESS_KEY_ID: a temporary limited scoped access key to 
+    Scipian's AWS account, which will be used by the state migrator to push
+    state to Scipian's backend
+    - SCIPIAN_AWS_SECRET_ACCESS_KEY: secret access key to the above account
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+No drawbacks have been identified with this feature.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This is an MVP level feature, allowing users to migrate state seamlessly if
+needed. A simple alternative manually move state, but that becomes labor
+intensive.
+
+As this feature is iterated on, it would likely become part of the self-service
+customer onboarding, which likely would include a UI. Some of this would then
+be abstracted away from the on-boarding customer. The internals will likely
+stay the same (i.e. a kubernetes job running a special Docker image to migrate
+state), however an onboarding customer wouldn't need to kick this off manually.
+
+# Prior art
+[prior-art]: #prior-art
+
+Terraform Enterprise offers some level of state migration, however the current
+use case for Scipian does not support Terraform Enterprise, and is therefore
+outside of the scope of this RFC.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+The following are open questions the Scipian dev community should answer before
+work commences:
+
+- What can be done to automate the creation of Scipian temporary limited access 
+AWS creds to distribute to on-boarding customers
+- Somewhat tied to this process is the question of how best to distribute the 
+Scipian Authentication CLI to onboarding teams?
+- What areas do we need to consider to align us for future iterations and
+improvements to automate this process?
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+As mentioned in this RFC, this is an MVP level solution to onboarding. This
```

Comment:
Should we add something here about how this is specific to importing Stack created infrastructure and that future iterations might support terraform more generally?